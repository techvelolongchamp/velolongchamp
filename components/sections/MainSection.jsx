import React from 'react'
import Image from 'next/image'
import PropTypes from 'prop-types'

import Background from '../../public/media/Banniere.JPG'

import { SectionContainer } from './MainSection.styled'

const MainSection = ({ id }) => {
  return (
    <SectionContainer id={id}>
      <Image
        src={Background}
        alt="Picture of the author"
        placeholder="blur"
        blurDataURL="LIE{hDW;9FMvcut7RPMxx^%MkBRP"
        layout="fill"
        objectFit="cover"
      />
    </SectionContainer>
  )
}

MainSection.propTypes = {
  id: PropTypes.string,
}

export default MainSection
