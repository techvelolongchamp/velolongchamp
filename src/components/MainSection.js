import React from 'react'
import styled from 'styled-components'

import Carousel from './Carousel'

const SectionContainer = styled.section`
  height: calc(100vh - ${({ theme }) => theme.navbarHeight});
  position: relative;
`

const MainContainer = styled.div`
  position: absolute;
  top: 100px;
  left: calc((100vw - 750px) / 2);
  z-index: 2;
`

const Content = styled.div`
  max-width: 750px;
  margin: 0 auto;
  color: ${({ theme }) => theme.colors.white};
`

const Title = styled.h1`
  font-size: 60px;
  font-weight: 300;
`

const SubTitle = styled.p`
  font-size: 20px;
`

const MainSection = () => {
  return (
    <SectionContainer>
      <Carousel />
      <MainContainer>
        <Content>
          <Title>Bienvenue sur le site de l'Association Vélo Longchamp</Title>
          <SubTitle>
            L'association Vélo Longchamp regroupe l'ensemble des cyclistes utilisateurs de la piste cyclable autour de l'hippodrome de
            Longchamp
          </SubTitle>
          <SubTitle>
            Elle a pour objectif entre autres de défendre les intérêts des usagers, promouvoir la pratique du vélo en Ile de France et un
            point d'information pour l'ensemble des usagers de la piste.
          </SubTitle>
        </Content>
      </MainContainer>
    </SectionContainer>
  )
}

export default MainSection
