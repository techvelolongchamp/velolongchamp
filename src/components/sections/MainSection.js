import React from 'react'
import styled from 'styled-components'

import Carousel from '../Carousel'

const SectionContainer = styled.section`
  min-height: 600px;
  height: calc(100vh - ${({ theme }) => theme.navbarHeight});
  position: relative;
  @media (max-width: 640px) {
    height: calc(100vh - ${({ theme }) => theme.mobielNavbarHeight});
  }
`

const MainContainer = styled.div`
  position: absolute;
  top: 100px;
  left: 0;
  right: 0;
  z-index: 2;
`

const Content = styled.div`
  max-width: 750px;
  width: calc(100vw - 80px);
  margin: 0 auto;
  padding: ${({ theme }) => theme.spacing.m};
  background-color: ${({ theme }) => theme.colors.brandSecondary}66;
  border-radius: 10px;
  color: ${({ theme }) => theme.colors.white};
`

const Title = styled.h1`
  font-size: ${({ theme }) => theme.fonts.xxlarge};
  font-weight: 300;
  @media (max-width: 640px) {
    font-size: ${({ theme }) => theme.fonts.xlarge};
  }
`

const SubTitle = styled.p`
  font-size: ${({ theme }) => theme.fonts.large};
  padding: ${({ theme }) => theme.spacing.xs} 0;
`

const MainSection = ({ id }) => {
  return (
    <SectionContainer id={id}>
      <Carousel />
      <MainContainer>
        <Content>
          <Title>Bienvenue sur le site de l'Association Vélo Longchamp</Title>
          <SubTitle>
            Vélo Longchamp regroupe l'ensemble des cyclistes utilisateurs de la
            piste cyclable autour de l'hippodrome de Longchamp
          </SubTitle>
          <SubTitle>
            Elle a pour objectif de défendre et promouvoir l’anneau de Longchamp
            pour que toutes et tous, nous puissions faire du vélo dans les
            meilleures conditions.
          </SubTitle>
        </Content>
      </MainContainer>
    </SectionContainer>
  )
}

export default MainSection
