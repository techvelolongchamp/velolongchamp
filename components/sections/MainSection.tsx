import React from 'react'
import Image from 'next/image'

import Background from '../../public/media/Banniere.JPG'

import { SectionContainer } from './MainSection.styled'

const MainSection: React.FC<{ id?: string }> = ({ id }) => {
  return (
    <SectionContainer id={id}>
      <Image
        src={Background}
        alt="Picture of the author"
        placeholder="blur"
        blurDataURL="LIE{hDW;9FMvcut7RPMxx^%MkBRP"
        layout="fill"
        objectFit="cover"
      />
    </SectionContainer>
  )
}

export default MainSection
