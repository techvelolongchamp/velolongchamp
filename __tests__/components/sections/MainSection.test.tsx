import React from 'react'
import { RenderWithProvider as render } from '../../provider'

import MainSection from '../../../components/sections/MainSection'

jest.mock('next/image', () => ({
  __esModule: true,
  default: (props: any) => {
    const keys = Object.keys(props)
    const simplifiedProps = {}
    keys.forEach((key) => {
      simplifiedProps[key.toLowerCase()] = props[key]
    })
    // eslint-disable-next-line @next/next/no-img-element, jsx-a11y/alt-text
    return <img {...simplifiedProps} />
  },
}))

describe('MainSection unit tests', () => {
  it('should render correctly', () => {
    const { container } = render(<MainSection />)

    expect(container).toMatchSnapshot()
  })
})
