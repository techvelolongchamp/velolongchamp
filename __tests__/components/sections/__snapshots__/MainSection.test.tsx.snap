// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainSection unit tests should render correctly 1`] = `
.c0 {
  min-height: 600px;
  height: calc(100vh - 80px);
}

@media (max-width:640px) {
  .c0 {
    height: calc(100vh - 110px);
  }
}

<div>
  <section
    class="c0"
  >
    <img
      alt="Banniere.JPG"
      blurdataurl="data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/4gIoSUNDX1BST0ZJTEUAAQEAAAIYAAAAAAQwAABtbnRyUkdCIFhZWiAAAAAAAAAAAAAAAABhY3NwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQAA9tYAAQAAAADTLQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAlkZXNjAAAA8AAAAHRyWFlaAAABZAAAABRnWFlaAAABeAAAABRiWFlaAAABjAAAABRyVFJDAAABoAAAAChnVFJDAAABoAAAAChiVFJDAAABoAAAACh3dHB0AAAByAAAABRjcHJ0AAAB3AAAADxtbHVjAAAAAAAAAAEAAAAMZW5VUwAAAFgAAAAcAHMAUgBHAEIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAFhZWiAAAAAAAABvogAAOPUAAAOQWFlaIAAAAAAAAGKZAAC3hQAAGNpYWVogAAAAAAAAJKAAAA+EAAC2z3BhcmEAAAAAAAQAAAACZmYAAPKnAAANWQAAE9AAAApbAAAAAAAAAABYWVogAAAAAAAA9tYAAQAAAADTLW1sdWMAAAAAAAAAAQAAAAxlblVTAAAAIAAAABwARwBvAG8AZwBsAGUAIABJAG4AYwAuACAAMgAwADEANv/bAEMAFA4PEg8NFBIQEhcVFBgeMiEeHBwePSwuJDJJQExLR0BGRVBac2JQVW1WRUZkiGVtd3uBgoFOYI2XjH2Wc36BfP/bAEMBFRcXHhoeOyEhO3xTRlN8fHx8fHx8fHx8fHx8fHx8fHx8fHx8fHx8fHx8fHx8fHx8fHx8fHx8fHx8fHx8fHx8fP/AABEIAZkCZgMBIgACEQEDEQH/xAAYAAEBAQEBAAAAAAAAAAAAAAAAAQQDBf/EABQQAQAAAAAAAAAAAAAAAAAAAAD/xAAXAQEBAQEAAAAAAAAAAAAAAAAAAQID/8QAFBEBAAAAAAAAAAAAAAAAAAAAAP/aAAwDAQACEQMRAD8A8cBzUAAQAABABFAAQAQAARUQAAQABFQAAEAUQAEABEVFQRUARUUQBRAAQEFEVAEVAEVBQABAAAFAAAAAAUAQVAFAUABAABUUAAAAHoAMKAAAAAAACCKiKACAACKiAACAAIqAAgAIoAAiKgCKioIqAIqKIAoiKgCKgoioAioAioKAAgAoAAAAAAACgAACKIoACoAAKgCgAAA9ABhQAAAQEAVAAARQAQEAAAEVEAEAAAQAEVFBFQBFQBAVBABAFERUUEVAEVBUAAQAEAUQAAAABQRQAAAAFRRAABUAUBQAEAAAAAAegAwAAAAAAAAAIgAAAACAACAgAAgAIAAogAIAogIIAigioAgKCACACoCAAgAAqAAAgKgCgAAAKIoAAKICKAAqCigCAAAAAAPQEGBUAAAAAAAABAEAAAAAEAAEAAAQAEABAAQFBAEEBRAQAEUEVBRFQBAAQAEAUEAAAABQQBQAAAURRAABUAUBQAEAAUQBRAHoCDAogCiAKIAAAAAAiCoAAAAgACAAAIAAIACAAigCAIqKCAIgCiAgoCAIqAIqAAgoAAgCgAAAAAKIAoAiiAKIoCoKKICKIAoAAAN4gyAAACAAAAACAogCoCAAAIAAAIAAIAACAigACAgAIoIqCCAoIIKAgAIACAAgoAAICqIAogCiAKAAqAiiKAACiAKAqAAAAAANwDIAAAAAIAAAgCiAAAAIgogACAqAACAAiioICoIACAAigCCCAqiACAgAIACAAgoAAIAogKogCiKAAAqAKAIKgCiAKAIAKAAAANwgyKIAogCiAKIIKIAqAACAogCoAAgCoAAgACAAgAIACKAIACCCAKIIoAgAIACAAgoAoCAKICqIAoigAAKgiKIoCoAogCgCACgqAKIA2gMgAAAAAAAAAgCAKIAqAACAogACAqAAIAAgAIoAgAIAioAgKCCAAgCAAgKCAKAgKIAogCiAKIooAIKgCiKAACiAKIIigKAAAANogyiiAKIAogKogCiAKIIAAAICiAKgACAAACAACKAIAIACACAAgigCAIACCKKgAIAACqAAAAAAAAAIKIAoigAAKgIogCiAKIAoANgDKAAAAAICiAKICqIAogCiAKIAAgKIAAgKggKgAIACAAggKgigIAIAggiqAgAIoqAAAAqAKIAogCiAKAAAKAAACACCiAKIAogCiANogyiiAKIAogCiAKIAogAAKAACAKIAAgKIACAAIACAAgAIoIAgggAIqgIACKAAAAAAAAAAAAAAAAAAAAAAAACoAogCiAKIA2CDDKiAqiAKIAogCiAKIAogCiAKICgICiAAICoACAoIACAAggioIAIKCAKCCgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADWIMsqIAogCiAKICqIAogCoAAAAICiAqoAAgACAAgAIIAgAgoIACAAgKoAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADSAyyAAAAAAAAACgAAAAgCiAKgACAKgAIAAgAgKCAAggACggCgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAANADLIAAAAAAAAAAAAAKCAKgAAgKIAAgAIoqCAqCAqCAAKCAKAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA7gIwAAACgCAAAAAAAIAogAAACCgCggAAgAIACKAACAKAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA7gIwAAAAAAIqCgAACAAAIKAAAIAAAgAAgoAogACKgoAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADuAjAAAAKAAgAgAKAgAAAACAAAAgAAgAAogKAAqAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA7gIyAAAAAAgAAACKgAAAIAAAioAACACgIAAoIqCgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAO4CMgAAAAAIAAAAioAAAgCAAqAAAgAAoioAAqoAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADuAjIAAAAAAioAAAioAAIgACKgoAAioAAKIqAAKqAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA0IojKCgIAAAAigIAAioAAIgACKgoAAioAAKIAACqgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAANACMgACKgAAAAIAAACACIAAioKAAIqAACoAAAqoAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADQAMgCAACAAAAgAAAIAIgACKiKAAIqKAAqAAAKqAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA/9k="
      layout="fill"
      objectfit="cover"
      placeholder="blur"
      src="test-file-stub"
    />
  </section>
</div>
`;
